mmengine/registry/registry.py

```diff
@@ -346,11 +381,29 @@ def get(self, key: str) -> Optional[Type]:
         obj_cls = None
         registry_name = self.name
         scope_name = self.scope
+
+        if scope is not None:
+            # import the registry to add the nodes into the registry tree
+            try:
+                import_module(f'{scope}.registry')
+                print_log(
+                    f'auto import {scope}.registry',
+                    logger='current',
+                    level=logging.DEBUG)
+            except (ImportError, AttributeError, ModuleNotFoundError):
+                print_log(
+                    f'Cannot auto import {scope}.registry, please check '
+                    f'whether the package "{scope}" is installed correctly '
+                    f'or import the registry manually.',
+                    logger='current',
+                    level=logging.DEBUG)
+        # lazy import the modules to register them into the registry
+        self.import_from_location()
+
```

Comment:
Should this code snippet be added to line

https://github.com/open-mmlab/mmengine/pull/643/files#diff-df9a9f722ab32299254fad7406178ad6e0a9e71dbc689d11af151999542f1ec2L365

and clean the code of `register_all_modules`